# Experiment e59f6f8e754071639637856c335eb955b565a2e313021be853f7604814a0def1
# Base 54ede352c4a6c628edb7e2b75f9bff85d8aa32ea
diff --git a/src/Panel.cpp b/src/Panel.cpp
index de1fc18b7..60ee4a1dc 100644
--- a/src/Panel.cpp
+++ b/src/Panel.cpp
@@ -1063,11 +1063,12 @@ long TPanel::wnd_proc(void* hwnd, unsigned int msg, unsigned int wParam, long lP
             return 0;
         }
 
-        target = this;
         if (panel_system.keyboardOwner() != 0) {
             target = panel_system.keyboardOwner();
         } else if (panel_system.modalPanel() != 0) {
             target = panel_system.modalPanel();
+        } else {
+            target = this;
         }
 
         if (msg != 0x100 && msg != 0x104) {
